src/main/drivers/barometer/barometer_bmp388.c

```diff
@@ -0,0 +1,414 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * BMP388 Driver author: Dominic Clifton
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+
+#include "build/build_config.h"
+
+#if defined(USE_BARO) && (defined(USE_BARO_BMP388) || defined(USE_BARO_SPI_BMP388))
```

Comment:
Why is this checking `defined(USE_BARO)`? Do you see any scenario where this driver will be used without BARO support, or should this be handled properly in `common_post.h`? Also, for what I can tell, there is no code that is exclusive to `defined(USE_BARO_BMP388)`, so a cleaner way of doing this would be to define `USE_BARO_BMP388` if `USE_BARO_SPI_BMP388` is defined, and then only check `USE_BARO_BMP388` here.

Reply:
@MJ666 what do you think?

Reply:
@mikeller can we get this merged as-is, i'm loathed to add more changes to this PR at this time.  Once a pattern as agreed on I can do it in a new PR for all the drivers.